### PR TITLE
fixed test falkiness

### LINKED ISSUE #331: Fix API Test Flakiness on CI (`bun test` workflow)
I have no idea why it's flaky. Good luck

/bounty $30

### PR DESCRIPTION
**Pull request by ShiboSoftwareDev**
/claim #331
/closes #331

the random port generation is not so random, but why does it cause issues I thought these tests ran synchronously

### CODE REVIEW

File: bun-tests/fake-snippets-api/fixtures/get-test-server.ts [hunk 1]
```diff
@@ -13,7 +15,7 @@ interface TestFixture {
 }
 
 export const getTestServer = async (): Promise<TestFixture> => {
-  const port = 3001 + Math.floor(Math.random() * 999)
+  const port = 3001 + testNumber++
```
Comment (seveibar) [inline bun-tests/fake-snippets-api/fixtures/get-test-server.ts:18]: can you use `get-port`/ `getPort()` instead?

Comment (seveibar) [reply bun-tests/fake-snippets-api/fixtures/get-test-server.ts:18]: honestly this isn't a bad approach too, maybe combine them- i think `getPort` allows you to specify a preferential port

**Review by seveibar — APPROVED** [review]

File: bun-tests/fake-snippets-api/fixtures/get-test-server.ts (hunk 1 above)
Comment (ShiboSoftwareDev) [reply bun-tests/fake-snippets-api/fixtures/get-test-server.ts:18]: @seveibar the random calculation as before but we pass it to getPort and it checks whether the port is used or not?

Comment (seveibar) [reply bun-tests/fake-snippets-api/fixtures/get-test-server.ts:18]: no sorry, your `testNumber++` approach passed to `getPort(...)` is my rec

**Comment by ShiboSoftwareDev** [comment]
with getPort():
image
without it:
image
@seveibar

**Comment by ShiboSoftwareDev** [comment]
you can notice it pauses a bit after each test

**Comment by seveibar** [comment]
Wow thats crazy, ok lets not use it- i had no idea

**Review by seveibar — APPROVED** [review]
Approved w/o getPort, nice investigation /tip $5

**Pull request merged**